Logger/BaseLoggerMixins.cs

```diff
@@ -1,6 +1,55 @@
-﻿namespace Logger;
-
-public static class BaseLoggerMixins
-{
-
-}
+﻿using System;
+
+namespace Logger;
+
+public static class BaseLoggerMixins
+{
+    public static void Error(this BaseLogger logger, string message, params object[] arguments)
+    {
+        if (logger != null)
+        {
+            logger.Log(LogLevel.Error, message);
+        }
+        else
+        {
+            throw new System.ArgumentNullException(null, "Error mixin got null");
+        }
+        
+    }

```

Comment:
```suggestion
    public static void Error(this BaseLogger logger, string message, params object[] arguments)
    {
        if (logger != null)
        {
            logger.Log(LogLevel.Error, message);
        }
        else
        {
            throw new System.ArgumentNullException(null, "Error mixin got null");
        }
        
    }

```
Have you considered using a helper method which can be called inside of each of these methods to check for null?

Reply:
+1 on this